### PR TITLE
Javascript/javascript2/week2

### PR DESCRIPTION
**Pull request by Poorooshka**
The first file which is seen as "commit" is in fact the "doubling" exercise.
The second file includes both html and javascript files related to "movies" exercises.

### CODE REVIEW

File: javascript/javascript2/week2/doubling.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title></title>
+
+    </head>
+
+    <body>
+
+
+        <script type="text/javascript">
```
Comment (benna100) [inline javascript/javascript2/week2/doubling.html:13]: Dont write your js in html!

File: javascript/javascript2/week2/doubling.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            let numbers = [1, 2, 3, 4];
+            let newNumbers = [];
+            let doubledNumbers = [];
+
+           
+
+            const oddNumber = number=> number % 2 !== 0;
+            const double = oddNum=> { console.log("The doubled numbers are", oddNum*2);}
+            newNumbers = numbers.filter(oddNumber);
```
Comment (benna100) [inline javascript/javascript2/week2/doubling.html:22]:
```suggestion
            const newNumbers = numbers.filter(oddNumber);
```

Then delete the newNumbers array above

File: javascript/javascript2/week2/main.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+const productsUl = document.querySelector('section.products ul');
+console.log(productsUl);
+
+function renderProducts(products) {
+        
+         while (productsUl.hasChildNodes()) {
+        productsUl.removeChild(productsUl.firstChild);
+      }
+        
+        products.forEach((product)=>{const li = document.createElement('li');
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:14]: Indentation very important!

File: javascript/javascript2/week2/main.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+"use strict"
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector('section.products ul');
+console.log(productsUl);
+
+function renderProducts(products) {
+        
+         while (productsUl.hasChildNodes()) {
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:10]: This seems a bit weird to me. Why not just do the forEach?

File: javascript/javascript2/week2/main.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        `;
+        productsUl.appendChild(li);});
+}
+
+
+
+// Search products by inputting names
+
+
+document.getElementById("searched").addEventListener('input', (event)=>{
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:35]:
```suggestion
document.querySelector("#searched").addEventListener('input', (event)=>{
```

File: javascript/javascript2/week2/main.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    const searchedItem = document.getElementById("searched").value.toLowerCase();
+    const searchedProducts = products.filter((product)=>product.name.toLowerCase().includes(searchedItem))
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:37]: very nice!

File: javascript/javascript2/week2/main.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    console.log(searchedProducts)
+    renderProducts(searchedProducts); 
+})
+    
+
+//Search Products by max price 
+document.getElementById("price").addEventListener('input', (event)=>{
+    const searchedPrice = document.getElementById("price").value;
+    const maxPrice= parseInt(searchedPrice);
+    const lessPriceProducts = products.filter((product)=>product.price < maxPrice);
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:47]: Very good variablenames here! Keep up the good work

File: javascript/javascript2/week2/movies.html [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+//Create an array of movie titles with long movie titles (titles bigger than 40)
+
+let longTitleMovies = [];
+longTitleMovies = movies.filter(movie=>movie.title.length > 40)
+console.log(longTitleMovies)
+
+//Count the number of movies made between 1980-1989 (including both the years)
+
+let newArray = [];
+newArray = movies.filter(movie=> 1980<movie.year<1989)
```
Comment (benna100) [inline javascript/javascript2/week2/movies.html:33]:
```suggestion
const eightiesMovies= movies.filter(movie=> 1980<movie.year<1989)
```

**Review by benna100 — CHANGES_REQUESTED** [review]
Looks good overall! Variablenames and indentation

File: javascript/javascript2/week2/doubling.html (hunk 2 above)
Comment (Poorooshka) [reply javascript/javascript2/week2/doubling.html:22]: I changed the whole code to the following 3 lines. Should I delete this file on github?

`const numbers = [1, 2, 3, 4];
const oddNumbers = numbers.filter(oddNum => oddNum % 2 !== 0)
oddNumbers.forEach( num=>console.log("The doubled numbers are "+ 2*num))`

Comment (Poorooshka) [reply javascript/javascript2/week2/doubling.html:22]: I changed the whole code to the following 3 lines. Should I delete this file on github?

const numbers = [1, 2, 3, 4]; 
const oddNumbers = numbers.filter(oddNum => oddNum % 2 !== 0) 
oddNumbers.forEach( num=>console.log("The doubled numbers are "+ 2*num))

File: javascript/javascript2/week2/main.js (hunk 5 above)
Comment (Poorooshka) [reply javascript/javascript2/week2/main.js:35]: why should I use querySelector instead?

File: javascript/javascript2/week2/doubling.html (hunk 2 above)
Comment (benna100) [reply javascript/javascript2/week2/doubling.html:22]: No need to delete. Just change the file and the add the change, commit it and push it

File: javascript/javascript2/week2/main.js (hunk 5 above)
Comment (benna100) [reply javascript/javascript2/week2/main.js:35]: You dont have to. But its good practice. The reason is that then using `getElementById` you need to give all elements you need to interact with an id. Id's should rarely be used in html. They are for very unique styles. 

querySelector is also consisten with css selectors so the selectors are the same no matter if you use css or js

**Pull request merged**